### PR TITLE
[iDEAL] Added pooling logic for iDEAL transactions

### PR DESCRIPTION
**Pull request by mpetrenco**

### CODE REVIEW

File: Source/Managers/IDEALManager.m [hunk 1]
```diff
@@ -85,7 +86,51 @@ - (void)getRedirectURLForIDEALBank:(IDEALBank *)iDealBank
 - (void)poolTransactionStatusForOrderId:(NSString *)orderId
                                checksum:(NSString *)checksum
                               completion:(JudoPoolingCompletion)completion {
-    // TODO: Handle pooling logic
+    
+
```
Comment (stpopa) [inline Source/Managers/IDEALManager.m]: extra line

File: Source/Managers/IDEALManager.m [hunk 2]
```diff
@@ -85,7 +86,51 @@ - (void)getRedirectURLForIDEALBank:(IDEALBank *)iDealBank
...
+    
+    [self getStatusForOrderId:orderId checksum:checksum completion:completion];
+    
+}
+
+- (void)getStatusForOrderId:(NSString *)orderId
+                   checksum:(NSString *)checksum
+                 completion:(JudoPoolingCompletion)completion {
+    
+    [self.session requestWithMethod:@"POST"
```
Comment (stpopa) [inline Source/Managers/IDEALManager.m]: why not `self.session POST:` ?

File: Source/Managers/IDEALManager.m [hunk 3]
```diff
@@ -85,7 +86,51 @@ - (void)getRedirectURLForIDEALBank:(IDEALBank *)iDealBank
...
+            return;
+        }
+        
+        if ([response.items.firstObject.orderDetails.orderStatus isEqual:@"FAIL"]) {
+            completion(IDEALStatusFailed, nil);
+            return;
+        }
+        
+        if ([response.items.firstObject.orderDetails.orderStatus isEqual:@"PENDING"]) {
+            sleep(10);
```
Comment (stpopa) [inline Source/Managers/IDEALManager.m]: let's discuss the logic here offline

File: Source/Managers/IDEALManager.m (hunk 2 above)
Comment (mpetrenco) [reply Source/Managers/IDEALManager.m]: The current POST implementation uses `path` as a parameter. This path is appended to the endpoint to generate a full URL that will be sent.

As this current request contains a completely different endpoint and changing the POST implementation might break something in other payment methods, it was easier to extract the request logic in a separate method.

**Pull request merged**